src/library/scala/collection/mutable/TreeMap.scala

```diff
@@ -133,12 +146,14 @@ sealed class TreeMap[K, V] private (tree: RB.Tree[K, V])(implicit val ordering:
     override def get(key: K) = if (isInsideViewBounds(key)) RB.get(tree, key) else None
 
     override def iterator = RB.iterator(tree, from, until)
+    override def keysIterator: Iterator[K] = RB.keysIterator(tree, from, until)
+    override def valuesIterator: Iterator[V] = RB.valuesIterator(tree, from, until)
```

Comment:
test isEmpty and return Iterator.empty?

Reply:
Added a commit along these lines. Right here, we can just cheaply test if the tree from which we are projecting a range is empty, and use `Iterator.empty`.